dali/kernels/erase/erase_cpu.h

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_ERASE_ERASE_CPU_H_
+#define DALI_KERNELS_ERASE_ERASE_CPU_H_
+
+#include <vector>
+#include <utility>
+#include "dali/core/common.h"
+#include "dali/core/convert.h"
+#include "dali/core/format.h"
+#include "dali/core/error_handling.h"
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/erase/erase_args.h"
+#include "dali/kernels/common/utils.h"
+
+namespace dali {
+namespace kernels {
+
+namespace detail {
+
+template <typename T, int Dims>
+void EraseKernelImpl(T *data,
+                     const TensorShape<Dims> &strides,
+                     const TensorShape<Dims> &shape,
+                     const T* fill_values,
+                     int channels_dim,
+                     std::integral_constant<int, 1>) {
+  assert(fill_values != nullptr);
+  for (int i = 0; i < shape[Dims - 1]; i++) {
+    data[i] = *fill_values;
+    if (channels_dim == Dims - 1) {
+      fill_values++;
```

Comment:
If `fill_values = &default_fill_value` then this line is wrong I guess.

Reply:
In this case channels_dim is `-1` (in purpose) so we will never enter this condition